src/components/CommentsList.jsx

```diff
@@ -1,33 +1,35 @@
-import { useEffect, useState } from "react";
+import { useContext, useEffect, useState } from "react";
 import { getComments } from "../utils/api";
 import { useParams } from "react-router-dom";
 import Comment from "./Comment";
+import CommentAdder from "./CommentAdder";
 
-const CommentList = () => {
+const CommentList = ({setShow}) => {
     const {article_id} = useParams();
     const [comments, setComments] = useState();
-    const [isLoading, setIsLoading] =useState(true);
+    const [isLoading, setIsLoading] = useState(true);
 
     useEffect(() => {
         getComments(article_id).then((res) => {
             setComments(res.data.comments)
             setIsLoading(false)
         })
-    }, []);
+    }, [comments]);
```

Comment:
Ah ok, the reason I did this is because when a user posts a new comment I'm optimistically rendering it, so it doesn't actually have an id in the database yet. So I realised a delete request to a newly posted comment would always fail unless the user manually refreshed the page, which I didn't really like. So this was just my solution for replacing that 'fake' comment with the real one without needing a refreash. Is there a better solution for doing this because it did stump me for a while?